qtfred/src/ui/dialogs/LoadoutDialog.cpp

```diff
@@ -237,7 +238,19 @@ LoadoutDialog::LoadoutDialog(FredView* parent, EditorViewport* viewport)
 }
 
 // a result of competing CI requirements
-LoadoutDialog::~LoadoutDialog(){} // NOLINT
+LoadoutDialog::~LoadoutDialog() {} // NOLINT
+
+void LoadoutDialog::closeEvent(QCloseEvent* e)
+{
+	if (!rejectOrCloseHandler(this, _model.get(), _viewport)) {
+		e->ignore();
+	};
+}
+
+void LoadoutDialog::rejectHandler()
+{
+	this->close();
+}
```

Comment:
Not sure what your asking here?

Reply:
All the changed dialogs were using effectively the same function just checking the _modifed variable in the model. So moveing the function to utils and just calling that massively reduces redundant code. The ones that weren't changed either had their own custom close event i.e the event editor. or are just shells with no function i.e. the briefing editor. I believe it was cyborg's desire to have all dialogs ask for confirmation on close or reject.

Reply:
Oh I completely agree with using a common function in the utils, as well as asking for confirmation when the dialog closes.  In fact I added close confirmation to several dialogs in legacy FRED.

What I was trying to say yesterday was this:
1. Do all the dialogs which need these functions have them?  (It looks like the answer is yes)
2. Can these functions be moved to a superclass?  (It looks like the answer is no, since the superclass is QDialog which we can't modify)

So we're good to go.